### ISSUE TITLE
field_reassign_with_default false positive with private fields

### ISSUE THREAD

**Issue by djc** [rust-lang/rust-clippy#6527]
Lint name: field_reassign_with_default

I tried this code:

```rust
    use http::uri;
    let mut parts = uri::Parts::default();
    parts.path_and_query = Some(uri::PathAndQuery::from_static(crate::DNS_QUERY_PATH));
    parts.scheme = Some(uri::Scheme::HTTPS);
    parts.authority = Some(
        uri::Authority::from_str(&name_server_name)
            .map_err(|e| ProtoError::from(format!("invalid authority: {}", e)))?,
    );
```

clippy diagnosed this with:

```rust
warning: field assignment outside of initializer for an instance created with Default::default()
  --> crates/https/src/request.rs:44:5
   |
44 |     parts.path_and_query = Some(uri::PathAndQuery::from_static(crate::DNS_QUERY_PATH));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::field_reassign_with_default)]` on by default
note: consider initializing the variable with `http::uri::Parts { path_and_query: Some(uri::PathAndQuery::from_static(crate::DNS_QUERY_PATH)), scheme: Some(uri::Scheme::HTTPS), authority: Some(
        uri::Authority::from_str(&name_server_name)
            .map_err(|e| ProtoError::from(format!("invalid authority: {}", e)))?,
    ), ..Default::default() }` and removing relevant reassignments
  --> crates/https/src/request.rs:43:5
   |
43 |     let mut parts = uri::Parts::default();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#field_reassign_with_default
```

However, `Parts` has a private `_priv` field, so functional record syntax to apply `Default::default()` doesn't work here.

### Meta

- `cargo clippy -V`: clippy 0.0.212 (e1884a8e 2020-12-29)
- `rustc -Vv`:
  ```
rustc 1.49.0 (e1884a8e3 2020-12-29)
binary: rustc
commit-hash: e1884a8e3c3e813aada8254edfa120e85bf5ffca
commit-date: 2020-12-29
host: x86_64-apple-darwin
release: 1.49.0
  ```

**Comment by alex-700** [comment]
duplicate of #6344 and #6515, fixed in #6375
